lib/zlib.js

```diff
@@ -95,6 +95,8 @@ function zlibBufferOnEnd() {
   var err;
   if (this.nread >= kMaxLength) {
     err = new errors.RangeError('ERR_BUFFER_TOO_LARGE');
+  } else if (this.nread === 0) {
+    buf = Buffer.alloc(0);
```

Comment:
I don't know if there is any performance penalty to use `Buffer.alloc(0)` vs `Buffer.allocUnsafe(0)`. However, as it's 0 I would use `allocUnsafe`.

Reply:
`alloc()` has what is essentially a fast path for `0`, `allocUnsafe()` doesn’t.

Reply:
This is very interesting. thanks!

Reply:
I mean, I don’t think it’s that way on purpose. :) It just happens to align nicely with how we can use basically `new Uint8Array(n)` for the `Buffer.alloc()` fast path, whereas `Buffer.allocUnsafe()` needs a bit of setup.